aptos-move/framework/src/module_metadata.rs

```diff
@@ -145,7 +145,16 @@ pub fn get_vm_metadata_v0(vm: &MoveVM, module_id: ModuleId) -> Option<RuntimeMod
 /// Extract metadata from a compiled module, upgrading V0 to V1 representation as needed.
 pub fn get_module_metadata(module: &CompiledModule) -> Option<RuntimeModuleMetadataV1> {
     if let Some(data) = find_metadata(module, &APTOS_METADATA_KEY_V1) {
-        bcs::from_bytes::<RuntimeModuleMetadataV1>(&data.value).ok()
+        let mut metadata = bcs::from_bytes::<RuntimeModuleMetadataV1>(&data.value).ok();
+        // Clear out metadata for v5, since it shouldn't have existed in the first place and isn't
```

Comment:
Can you leave a TODO here + file a bug in aptos_vm.rs to make sure we add a check for mismatching bytecode and metadata?

Reply:
I think it's already there if I'm not mistaken https://github.com/aptos-labs/aptos-core/blob/08ba4052c2e8b1de994706d1faf0f84e7c935ae1/aptos-move/aptos-vm/src/aptos_vm.rs#L675

Reply:
Yeah, it wasn't clear what he was asking for, so I just ignored it ;)

Reply:
Above is not that check, this just calls the general metadata verification code. I opened yesterday an issue which covers what we want here: #6420